### PR TITLE
feat(sdk): add command response dotnet sdk

### PR DESCRIPTION
**Pull request by dooriya**
UT will be added in another PR.

E2E TEST: N/A

### CODE REVIEW

File: packages/dotnet-sdk/src/TeamsFx/Conversation/ActivityCommandResponse.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        /// <exception cref="ArgumentNullException"><paramref name="activity"/> is null.</exception>
+        public ActivityCommandResponse(IActivity activity)
+        {
+            Activity = activity ?? throw new ArgumentNullException(nameof(activity));
+        }
+
+        /// <inheritdoc/>
+        public async Task SendResponseAsync(ITurnContext turnContext, CancellationToken cancellationToken = default)
+        {
+            await turnContext.SendActivityAsync(Activity, cancellationToken);
```
Comment (swatDong) [inline packages/dotnet-sdk/src/TeamsFx/Conversation/ActivityCommandResponse.cs:35]: `ConfigureAwait(false)` for all await operation.
(Fix all warnings/suggestions provided by VS itself.)

File: packages/dotnet-sdk/src/TeamsFx/Conversation/RegExpTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        /// <exception cref="ArgumentNullException"><paramref name="pattern"/> is null.</exception>
+        public RegExpTrigger(string pattern)
+        {
+            Pattern = pattern ?? throw new ArgumentNullException(nameof(pattern));
+        }
+
+        /// <inheritdoc/>
+        public bool ShouldTrigger(string input)
+        {
+            var regex = new Regex(Pattern, RegexOptions.IgnoreCase);
```
Comment (swatDong) [inline packages/dotnet-sdk/src/TeamsFx/Conversation/RegExpTrigger.cs:33]: How about directly accepting `Regex` from user? To let caller to determine options and timeout.
E.g., `ITriggerPattern` has one method `ShouldTrigger`. Then, `StringTrigger` takes `string` as input, while `RegexTrigger` takes `Regex` as input.

File: packages/dotnet-sdk/src/TeamsFx/Conversation/ActivityCommandResponse.cs (hunk 1 above)
Comment (dooriya) [reply packages/dotnet-sdk/src/TeamsFx/Conversation/ActivityCommandResponse.cs]: Good catch and already fixed.

File: packages/dotnet-sdk/src/TeamsFx/Conversation/RegExpTrigger.cs (hunk 2 above)
Comment (dooriya) [reply packages/dotnet-sdk/src/TeamsFx/Conversation/RegExpTrigger.cs]: Agree, that would be more flexible for developers.

File: packages/dotnet-sdk/src/TeamsFx/Conversation/CommandTriggerType.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.TeamsFx.Conversation
+{
+    /// <summary>
+    /// Type of trigger pattern used to trigger a <see cref="ITeamsCommandHandler"/> instance.
+    /// </summary>
+    public enum CommandTriggerType
```
Comment (swatDong) [inline packages/dotnet-sdk/src/TeamsFx/Conversation/CommandTriggerType.cs:9]: So, is this trigger type redundant now?

Comment (dooriya) [reply packages/dotnet-sdk/src/TeamsFx/Conversation/CommandTriggerType.cs:9]: Yeah~ Let me remove it later.

**Review by swatDong — APPROVED** [review]

**Pull request merged**